osu.Game/Rulesets/Mods/ModNightcore.cs

```diff
@@ -34,5 +43,47 @@ public override void ApplyToTrack(Track track)
             track.AddAdjustment(AdjustableProperty.Frequency, freqAdjust);
             track.AddAdjustment(AdjustableProperty.Tempo, tempoAdjust);
         }
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
+        {
+            drawableRuleset.Overlays.Add(new NightcoreBeatContainer());
+        }
+
+        public class NightcoreBeatContainer : BeatSyncedContainer
+        {
+            private SkinnableSound hatSample;
+            private SkinnableSound clapSample;
+            private SkinnableSound kickSample;
+            private SkinnableSound finishSample;
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                InternalChildren = new Drawable[]
+                {
+                    hatSample = new SkinnableSound(new SampleInfo("nightcore-hat")),
+                    clapSample = new SkinnableSound(new SampleInfo("nightcore-clap")),
+                    kickSample = new SkinnableSound(new SampleInfo("nightcore-kick")),
+                    finishSample = new SkinnableSound(new SampleInfo("nightcore-finish")),
+                };
+            }
+
+            protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, TrackAmplitudes amplitudes)
+            {
+                base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
+
+                if (beatIndex > -1)
+                {
+                    if (beatIndex % 16 == 0)
+                        finishSample?.Play();
+                    else if (beatIndex % 2 == 0)
+                        kickSample?.Play();
+                    else if (beatIndex % 2 == 1)
+                        clapSample?.Play();
+                    else
```

Comment:
How can this be reachable after `%2 == 0` and `%2 == 1`?

Reply:
I've based off the logic in smoogipoo's reply here https://github.com/ppy/osu/issues/6060#issuecomment-548198973 however I never saw the hat being used even in stable so I'm not quite sure about it.